flocker/node/_deploy.py

```diff
@@ -934,6 +935,13 @@ def _restart_for_application_change(
         # we don't want to do anything:
         comparable_state = comparable_state.transform(["running"], True)
 
+        # Restart policies were briefly supported but they interact poorly with
```

Comment:
This comment is wrong, as is implementation I think. Given this patch Flocker will *not* change any existing running container; rather, it will continuously restart it.

1. Container X is configured with `always`.
2. Actual container may be `always`, may be new one with `never.
3. In either case, this function will indicate state is `never.
4. `always != never` and so container will be restarted (repeatedly).

What you want to change is the *configuration* to be `never` in this function. Then:

* Existing containers with policy `always` or `on-failure` in their state will be restarted as `never` containers, thus preventing problems on restart.
* Containers with state `never` will always compare equally to the configuration and therefore not getting restarted.
* When we re-introduce restarts the line can be deleted and the containers will be appropriately restarted.

In short: transform `comparable_configuration`, not `comparable_state`.